stackslib/src/net/api/tests/postblock_proposal.rs

```diff
@@ -0,0 +1,427 @@
+        >,
+    ) {
+        let mut results = self.results.lock().unwrap();
+        results.push(result);
+        self.condvar.notify_one();
+    }
+}
+
+struct ProposalTestObserver {
+    pub proposal_observer: Arc<Mutex<ProposalObserver>>,
+}
+
+impl ProposalTestObserver {
+    fn new() -> Self {
+        Self {
+            proposal_observer: Arc::new(Mutex::new(ProposalObserver::new())),
+        }
+    }
+}
+
+impl ProposalCallbackReceiver for Arc<Mutex<ProposalObserver>> {
+    fn notify_proposal_result(
+        &self,
+        result: Result<
+            postblock_proposal::BlockValidateOk,
+            postblock_proposal::BlockValidateReject,
+        >,
+    ) {
+        let observer = self.lock().unwrap();
+        observer.notify_proposal_result(result);
+    }
+}
+
+impl MemPoolEventDispatcher for ProposalTestObserver {
+    fn get_proposal_callback_receiver(&self) -> Option<Box<dyn mempool::ProposalCallbackReceiver>> {
+        Some(Box::new(Arc::clone(&self.proposal_observer)))
+    }
+
+    fn mempool_txs_dropped(&self, txids: Vec<Txid>, reason: mempool::MemPoolDropReason) {}
+
+    fn mined_block_event(
+        &self,
+        target_burn_height: u64,
+        block: &crate::chainstate::stacks::StacksBlock,
+        block_size_bytes: u64,
+        consumed: &ExecutionCost,
+        confirmed_microblock_cost: &ExecutionCost,
+        tx_results: Vec<crate::chainstate::stacks::miner::TransactionEvent>,
+    ) {
+    }
+
+    fn mined_microblock_event(
+        &self,
+        microblock: &StacksMicroblock,
+        tx_results: Vec<crate::chainstate::stacks::miner::TransactionEvent>,
+        anchor_block_consensus_hash: ConsensusHash,
+        anchor_block: BlockHeaderHash,
+    ) {
+    }
+
+    fn mined_nakamoto_block_event(
+        &self,
+        target_burn_height: u64,
+        block: &crate::chainstate::nakamoto::NakamotoBlock,
+        block_size_bytes: u64,
+        consumed: &ExecutionCost,
+        tx_results: Vec<crate::chainstate::stacks::miner::TransactionEvent>,
+    ) {
+    }
+}
+
+#[test]
+fn test_try_make_response() {
+    let addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), 33333);
+    let test_observer = TestEventObserver::new();
+    let mut rpc_test = TestRPC::setup_nakamoto(function_name!(), &test_observer);
+    let mut requests = vec![];
+
+    let tip =
+        SortitionDB::get_canonical_burn_chain_tip(&rpc_test.peer_1.sortdb.as_ref().unwrap().conn())
+            .unwrap();
+    let miner_privk = &rpc_test.peer_1.miner.nakamoto_miner_key();
+
+    let mut block = {
+        let chainstate = rpc_test.peer_1.chainstate();
+        let parent_stacks_header = NakamotoChainState::get_block_header(
+            chainstate.db(),
+            &tip.get_canonical_stacks_block_id(),
+        )
+        .unwrap()
+        .unwrap();
+
+        // let mut block = make_codec_test_nakamoto_block(StacksEpochId::Epoch30, &tip, miner_privk, parent_stacks_header);
```

Comment:
dead code

Reply:
Removed in d5dc115ebab9307b8955ee8656def6bdb6012287.